api/v1alpha1/osbuildconfig_types.go

```diff
@@ -0,0 +1,136 @@
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	buildv1 "github.com/openshift/api/build/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// OSBuildConfigSpec defines the desired state of OSBuildConfig
+type OSBuildConfigSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Details defines what to build
+	Details BuildDetails `json:"build_details"`
+	// Triggers defines when to build
+	Triggers BuildTriggers `json:"build_triggers"`
+}
+
+// BuildDetails includes all the information needed to build the image
+type BuildDetails struct {
+	// Distribution is the name of the O/S distribution
+	Distribution string `json:"distribution"`
+	// Customizations defines the changes to be applied on top of the base image (optional)
+	Customizations *Customizations `json:"customizations,omitempty"`
+	// TargetImage defines the requested output image
+	TargetImage TargetImage `json:"target_image"`
+}
+
+// Customizations defines the changes to be applied on top of the base image
+type Customizations struct {
+	// Packages is a list of RPM packages to install (optional)
+	Packages []string `json:"packages,omitempty"`
+	// Users is the list of Users to add to the image (optional)
+	Users []User `json:"users,omitempty"`
+	// Services defines the services to enable or disable (optional)
+	Services *Services `json:"services,omitempty"`
+}
+
+// User defines a single user to be configured
+type User struct {
+	// Name is the username for the new user
+	Name string `json:"name"`
+	// Groups is the groups to add the user to (optional)
+	Groups []string `json:"groups,omitempty"`
+	// Key is the user's SSH public key (optional)
+	PubKey *string `json:"pub_key,omitempty"`
+}
+
+type Services struct {
+	// Enabled is the list of services to enabled (optional)
+	Enabled []string `json:"enabled,omitempty"`
+	// Disabled is the list of services to disabled (optional)
+	Disabled []string `json:"disabled,omitempty"`
+}
+
+type TargetImage struct {
+	// Architecture defines target architecture of the image
+	Architecture Architecture `json:"architecture"`
+	// ImageType defines the target image type
+	TargetImageType TargetImageType `json:"target_image_type"`
+	// OSTree is the OSTree configuration of the build (optional)
+	OSTree *OSTreeConfig `json:"ostree,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=x86_64;aarch64
+type Architecture string
+
+// +kubebuilder:validation:Enum=edge-container;edge-installer
+type TargetImageType string
+
+// OSTreeConfig defines the OSTree ref details
+type OSTreeConfig struct {
+	// Url is the URL of the target build
+	Url *string `json:"url"`
+	// Ref is the ref of the target build
+	Ref *string `json:"ref"`
```

Comment:
any reason for not using `Reference` as the variable name? It might be more readable. WDYT?

Reply:
As I mentioned before, this structure is a copy of the one from the REST V2 API. Isn't `ref` the name of the parameter passed to the composer API?

Reply:
Following our discussion, I will leave it as `ref` for now as it used in both the Composer API and the composer-cli. We can change it later